pipelines/copy-number-variation-vector/gcp/HMM.wdl

```diff
@@ -0,0 +1,335 @@
+##
+
+
+workflow HMM {
+  String pipeline_version = "1.0.0"
+
+  input {
+    #windowed coverage parameters
+    File input_bam
+    File input_bai
+    String sample_id
+    String output_dir
+    Int interval
+    Int window_size
+    Int min_qual
+    #coverage summary parameters
+    Float accessibility_threshold
+    Float mapq_threshold
+    File accessibility_mask_file
+    File mapq_file
+    File gc_content_file
+    String sample_group_id
+    String species
+    String runtime_zones
+  }
+
+  call WindowedCoverage {
+    input:
+      input_bam = input_bam,
+      input_bai = input_bai,
+      sample_id = sample_id,
+      output_dir = output_dir,
+      interval = interval,
+      window_size = window_size,
+      min_qual = min_qual,
+      runtime_zones = runtime_zones
+  }
+
+  call CoverageSummary {
+    input:
+      coverage_tarball = WindowedCoverage.output_gz,
+      accessibility_threshold = accessibility_threshold,
+      mapq_threshold = mapq_threshold,
+      accessibility_mask_file = accessibility_mask_file,
+      mapq_file = mapq_file,
+      gc_content_file = gc_content_file,
+      sample_group_id = sample_group_id,
+      output_dir = output_dir,
+      sample_id = sample_id,
+      runtime_zones = runtime_zones
+  }
+
+  call CoverageHMM {
+    input:
+      coverage_tarball = CoverageSummary.output_gz,
+      gc_content_file = gc_content_file,
+      coverage_gc = CoverageSummary.coverage_output,
+      coverage_variance = CoverageSummary.variance_output,
+      mapq_file = mapq_file,
+      mapq_threshold = mapq_threshold,
+      species =  species,
+      output_dir = output_dir,
+      sample_id = sample_id,
+      runtime_zones = runtime_zones
+  }
+  
+  output {
+    File output_gz = CoverageHMM.output_gz
+    File coverage_variance = CoverageSummary.variance_output
+  }
+}
+
+task WindowedCoverage {
+  input {
+    File input_bam
+    File input_bai
+    String sample_id
+    String output_dir
+    Int interval
+    Int window_size
+    Int min_qual
+
+    # runtime values
+    String docker = "us.gcr.io/broad-gotc-prod/cnv:1.0.0-1679431881"
+    String ram = "8000 MiB"
+    Int cpu = 16
+    # TODO: Make disk space dynamic based on input size
+    Int disk = 70
+    Int preemptible = 3
+    String runtime_zones
+  }
+  meta {
+    description: "Compute aligned read counts across the genome in 300 bp windows. The output is a set of Windowed count reads file, 1 row per window, 1 file per sample - compressed in a tar.gz file to keep the directory structure."
+  }
+  parameter_meta {
+    input_bam: "The input BAM file"
+    input_bai: "The BAM file's corresponding index file"
+    sample_id: "The sample name. This is typically the same as the input bam file name but without the extension."
+    output_dir: "The output directory name. This is set to `coverage` by default."
+    interval: "The interval size (bp) to use for the coverage calculations. Default is 300."
```

Comment:
is `interval` , `window_size`, and `min_qual` actually defaulted somewhere? I'm not seeing them set to a default, but I could totally be missing it! Maybe defaulted in the input json somewhere?

Reply:
These are defaulted in the top level WDL

Reply:
oh yeah duh, i see them now 🤦‍♀️ :
```   
 Int interval = 300
    Int window_size = 300
    Int min_qual = 10